api/app/src/Controller/SelfRegisterController.php

```diff
@@ -109,7 +109,15 @@ public function verifyCoDeputy(Request $request, UserRegistrationService $userRe
 
         try {
             $coDeputyVerified = $userRegistrationService->validateCoDeputy($selfRegisterData);
-            $coDeputyUid = $userRegistrationService->retrieveCoDeputyUid();
+            $coDeputyUid = $userRegistrationService->retrieveCoDeputyUid($selfRegisterData);
+
+            $user = $this->em->getRepository('App\Entity\User')->findOneByEmail($selfRegisterData->getEmail());
+            if (!$user) {
+                throw new \RuntimeException('User registration: not found', 421);
+            }
+
+            // check if it's the primary account for the co-deputy
+            $existingDeputyAccounts = $this->em->getRepository('App\Entity\User')->findBy(['deputyUid' => $coDeputyUid]);
```

Comment:
Should this be only setting $existingDeputyAccounts to true if the findBy returns more than 1?

Reply:
The codeputies UID and other properties are set in the User table after this function returns values back to the CoDeputyController, so this line of code is checking to see if there are any deputy accounts that already exist.